### PR TITLE
Introduce a prometheus loader tool.

### PR DESCRIPTION
**Pull request by mrsiano**
the following tool allow us to generate some workloads against promethues specially for openhift stack.

motivation:
come up with a tool that allow us to simulate the grafana workloads.
the tool fire prometheus queries simultaneously over http against the prometheus server as the grafana dose

### CODE REVIEW

**Comment by mrsiano** [comment]
@jeremyeder @mffiedler

**Comment by mrsiano** [comment]
I've tested this tool and I'm already using it, I really can see it affect the prometheus performance.

**Comment by jeremyeder** [comment]
can you throw together a README about how to use it?
is @brancz ok with the approach?

**Comment by brancz** [comment]
Generally this approach seems fine. Just a few minor questions that could also be follow ups:

* How did you choose the queries? Ideally I’d like to use the queries that the Grafana dashboards generate (we merged a Grafana integration in the cluster monitoring stack a few days ago).
* While some constant load seems fine we shouldn’t overdo it with the concurrency when we do the actual tests. I didn’t check the queries super thoroughly but it’s for sure possible to kill a prometheus Server if you do a brutally complex query - like any other database. We should make sure it’s a good mix of being realistic and relatively constant. Remember that we already have constant query load because of recording rules, what we’re trying to simulate here is ad hoc (random query but rather rarely executed) and dashboard querying (typically dashboards reload data every 10s).

**Comment by jeremyeder** [comment]
just out of curiosity why would our default dashboard reload interval be any smaller than the scrape interval?  (or even 2x the scrape interval?)  obviously users can dial it down to something lower.

**Comment by mrsiano** [comment]
@brancz 
regarding your questions:
- those query I'v collected to the qs.txt file being collected from here x which is our grafana dashboard for several diffrent CI clusters.
- about the concurrency that how the grafana works, in case you have few graphs in your dashboard and you refresh it you will see simultaneous requests by any network tracer.
image
about the complexity it depends on what data you wanted to draw, in our case we developed some graphs that needs data aggregation and some ohter math functions (BTW, it's pretty much basic and not so complex).
any way we can tune that qs.file and add or remove that collection accordingly.

@jeremyeder 
currently the default is 20 sec, but the user can setup what ever he want and control it by the command line arg, I can hard code a minimum  value to guarantee the end use will not overkill it.

**Comment by brancz** [comment]
@mrsiano got it, all of that sounds good to me, thanks for clarifying.

@jeremyeder I'm not sure I understand how querying is related to scrape interval here. Can you elaborate what you meant?

**Comment by jeremyeder** [comment]
I meant from a practical standpoint as a user, that isn't it useless to have your dashboard refresh interval lower than the scrape interval?  No new data will show up?

**Comment by mrsiano** [comment]
@mffiedler can you merge this please ?

File: openshift_scalability/PrometheusLoader.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Intro
+this tools used to fire prometheus queries against prometheus instance in OpenShift
```
Comment (mffiedler) [inline openshift_scalability/PrometheusLoader.md]: Editorial:  "This tool is used to fire prometheus queries against a prometheus instance in OpenShift."

File: openshift_scalability/PrometheusLoader.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+stack context.
+
+It's also helps to simulate Grafana workloads, on staging cluster.
```
Comment (mffiedler) [inline openshift_scalability/PrometheusLoader.md]: Editorial:  "It also helps  to simulate Grafana workloads on a staging cluster."

File: openshift_scalability/PrometheusLoader.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+the app uses query file which include a collection of prometheus queries,
+the collection fits only to OpenShift context
+
```
Comment (mffiedler) [inline openshift_scalability/PrometheusLoader.md]: Editorial:  "The app uses a query file which includes a collection...."

**Comment by mrsiano** [comment]
@mffiedler any blocker here ?

**Pull request merged**